### PR TITLE
[issue 1458]reduce multi-buffer size

### LINKED ISSUE #1458: There is no need to add padding for the most recent buffer because there is no buffer after it.
There is https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1354 which directly modifies fdb in order to resolve #1349 without retuning. It blindly adds 3*255 bytes to workspace size, which should be enough to align 4 sub-buffers.

**_There is no need to add padding for the most recent buffer because there is no buffer after it._**

I suspect that the reason of the issue is that `MultiBufferWorkspaceTraits::MultiBufferWorkspaceTraits()` wants too much padding. It adds padding for all four buffers, and total padding may exceed 3*255.

I think that the proper fix should remove adding padding for the last sub-buffer. `MultiBufferWorkspaceTraits::GetSize()` should return offset of the beginning of the last sub-buffer + the size of the last sub-buffer. **_Basically, no other fixes required._**

_Originally posted by @atamazov in https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1429#issuecomment-1048305729_

### PR DESCRIPTION
**Pull request by shaojiewang**
Resolve #1458 
reduce multi buffer size by using offset of last buffer plus last buffer size

### CODE REVIEW

**Comment by shaojiewang** [comment]
Issue #1458 is originally posted by @atamazov . So also needs Artem's review

**Review by atamazov — DISMISSED** [review]
LGTM!

File: src/buffer_info.cpp [hunk 1]
```diff
@@ -162,7 +162,10 @@ MultiBufferWorkspaceTraits::MultiBufferWorkspaceTraits(std::initializer_list<siz
     }
 }
 
-size_t MultiBufferWorkspaceTraits::GetSize() const { return v_offset.back(); }
+size_t MultiBufferWorkspaceTraits::GetSize() const
+{
+    return v_offset[v_offset.size() - 2] + v_size.back();
```
Comment (atamazov) [inline src/buffer_info.cpp:167]: [Recommendation] Maybe you'll find this nicer
```suggestion
    return (&v_offset.back())[-1] + v_size.back();
```

**Review by atamazov — APPROVED** [review]
LGTM

**Review by carlushuang — APPROVED** [review]
LGTM

**Review by junliume — APPROVED** [review]

**Pull request merged**